src/3d/terrain/qgsterrainentity_p.cpp

```diff
@@ -132,6 +127,70 @@ bool QgsTerrainEntity::rayIntersection( const QgsRayCastingUtils::Ray3D &ray, QV
   return minDist >= 0;
 }
```

Comment:
does the previous intersection function still have a use?

Reply:
It's used in `Qgs3DAxis` logic

Reply:
Can it be replaced in `Qgs3DAxis` logic in order to remove the old function?

Reply:
I removed the old function and updated the 3d axis code to use the new one.
Also fixed a bug, as the previous implementation did not cater for potential terrain offset.

Reply:
Thanks @uclaros . I factored the terrain offset issue at https://github.com/qgis/QGIS/pull/52316